sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -780,27 +803,31 @@ class AccessPolicy(GenAccessPolicy):
         Required unless an id is given referencing a stored access policy
         which contains this field. This field must be omitted if it has been
         specified in an associated stored access policy.
-    :type permission: str or ~azure.storage.blob.ContainerSasPermissions
+    :type permission: Optional[Union[ContainerSasPermissions, str]]
     :param expiry:
         The time at which the shared access signature becomes invalid.
         Required unless an id is given referencing a stored access policy
         which contains this field. This field must be omitted if it has
         been specified in an associated stored access policy. Azure will always
         convert values to UTC. If a date is passed in without timezone info, it
         is assumed to be UTC.
-    :type expiry: ~datetime.datetime or str
+    :type expiry: Optional[Union[str, datetime]]
     :param start:
         The time at which the shared access signature becomes valid. If
         omitted, start time for this call is assumed to be the time when the
         storage service receives the request. Azure will always convert values
         to UTC. If a date is passed in without timezone info, it is assumed to
         be UTC.
-    :type start: ~datetime.datetime or str
+    :type start: Optional[Union[str, datetime]]
     """
-    def __init__(self, permission=None, expiry=None, start=None):
-        self.start = start
-        self.expiry = expiry
-        self.permission = permission
+    def __init__(
+        self, permission: Optional[Union["ContainerSasPermissions", str]] = None,
+        expiry: Optional[Union[str, "datetime"]] = None,
+        start: Optional[Union[str, "datetime"]] = None
+    ) -> None:
+        self.start = start  #type: ignore
+        self.expiry = expiry  #type: ignore
+        self.permission = permission  #type: ignore
```

Comment:
azure\storage\blob\_models.py:828: error: Incompatible types in assignment (expression has type "Union[str, datetime, None]", variable has type "Optional[str]")  [assignment]
azure\storage\blob\_models.py:829: error: Incompatible types in assignment (expression has type "Union[str, datetime, None]", variable has type "Optional[str]")  [assignment]
azure\storage\blob\_models.py:830: error: Incompatible types in assignment (expression has type "Union[ContainerSasPermissions, str, None]", variable has type "Optional[str]")  [assignment]

I think this is due to how these variables are defined in `GenAccessPolicy` and this inherits from that?

Reply:
This looks right now.

Reply:
![image](https://github.com/Azure/azure-sdk-for-python/assets/101599632/c5b7ba71-7899-4c31-92fd-c560e564d69c)
It's because of the generated code:
`class AccessPolicy(GenAccessPolicy):`
And our `AccessPolicy` has family ties 😢